sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFileDestinationCheckpointData.cs

```diff
@@ -1,16 +1,196 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.IO;
+using System.Text;
+using Azure.Core;
+using Azure.Storage.Files.Shares.Models;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
 
 namespace Azure.Storage.DataMovement.Files.Shares
 {
     internal class ShareFileDestinationCheckpointData : StorageResourceCheckpointData
     {
-        public override int Length => 0;
+        private const char HeaderDelimiter = Constants.CommaChar;
+
+        /// <summary>
+        /// Schema version.
+        /// </summary>
+        public int Version;
+
+        /// <summary>
+        /// The content headers for the destination blob.
+        /// </summary>
+        public ShareFileHttpHeaders ContentHeaders;
+        private byte[] _contentTypeBytes;
+        private byte[] _contentEncodingBytes;
+        private byte[] _contentLanguageBytes;
+        private byte[] _contentDispositionBytes;
+        private byte[] _cacheControlBytes;
+
+        /// <summary>
+        /// The metadata for the destination blob.
+        /// </summary>
+        public Metadata Metadata;
+        private byte[] _metadataBytes;
+
+        public override int Length => CalculateLength();
+
+        public ShareFileDestinationCheckpointData(
+            ShareFileHttpHeaders contentHeaders,
+            Metadata metadata)
+        {
+            Version = DataMovementShareConstants.DestinationCheckpointData.SchemaVersion;
+            ContentHeaders = contentHeaders;
+            _contentTypeBytes = ContentHeaders?.ContentType != default ? Encoding.UTF8.GetBytes(ContentHeaders.ContentType) : Array.Empty<byte>();
+            _contentEncodingBytes = ContentHeaders?.ContentEncoding != default ? Encoding.UTF8.GetBytes(string.Join(HeaderDelimiter.ToString(), ContentHeaders.ContentEncoding)) : Array.Empty<byte>();
+            _contentLanguageBytes = ContentHeaders?.ContentLanguage != default ? Encoding.UTF8.GetBytes(string.Join(HeaderDelimiter.ToString(), ContentHeaders.ContentLanguage)) : Array.Empty<byte>();
```

Comment:
nit: Doesn't `string.Join` have an overload for a `char` so you don't need `ToString` here?

Reply:
Not in netstandard2.0 it seems. Wouldn't compile.

Reply:
netstandard2.0 holding us back once again :(